src/main/java/grandduke/command/Parser.java

```diff
@@ -0,0 +1,203 @@
+package grandduke.command;
+
+import grandduke.task.TaskList;
+import grandduke.task.Deadline;
+import grandduke.task.Event;
+import grandduke.task.Task;
+import grandduke.task.Todo;
+import grandduke.exception.GrandException;
+import grandduke.exception.UnrecognisedCommandException;
+import grandduke.exception.deadline.EmptyDeadlineDateException;
+import grandduke.exception.deadline.EmptyDeadlineDescriptionException;
+import grandduke.exception.deadline.EmptyDeadlineException;
+import grandduke.exception.deadline.MissingByException;
+import grandduke.exception.event.EmptyEventDescriptionException;
+import grandduke.exception.event.EmptyEventException;
+import grandduke.exception.event.EmptyEventFromException;
+import grandduke.exception.event.EmptyEventToException;
+import grandduke.exception.event.MissingFromException;
+import grandduke.exception.event.MissingToException;
+import grandduke.exception.todo.EmptyTodoException;
+
+public abstract class Parser {
+    /**
+     * Parses the command from the user and decides on the appropriate command to
+     * execute
+     * 
+     * @param input
+     *              the command sent by the user
+     */
+    public static void parseCommand(String input) throws UnrecognisedCommandException {
+        String[] inputArr = input.split(" ", 2);
+        String command = inputArr[0];
+
+        String commandDetails;
+
+        if (inputArr.length > 1) {
+            commandDetails = inputArr[1].strip();
+        } else {
+            commandDetails = "";
+        }
+
+        try {
+            switch (command) {
+                case Io.LIST_COMMAND:
+                    TaskList.printTaskList();
+                    break;
+
+                case Io.MARK_COMMAND:
+                    TaskList.markTask(commandDetails, true);
+                    break;
+
+                case Io.UNMARK_COMMAND:
+                    TaskList.markTask(commandDetails, false);
+                    break;
+
+                case Io.TODO_COMMAND:
+                    TaskList.addTask(commandDetails, Io.TODO_COMMAND);
+                    break;
+
+                case Io.DEADLINE_COMMAND:
+                    TaskList.addTask(commandDetails, Io.DEADLINE_COMMAND);
+                    break;
+
+                case Io.EVENT_COMMAND:
+                    TaskList.addTask(commandDetails, Io.EVENT_COMMAND);
+                    break;
+
+                case Io.DELETE_COMMAND:
+                    TaskList.deleteTask(commandDetails);
+                    break;
+
+                default:
+                    throw new UnrecognisedCommandException();
+            }
+        } catch (GrandException e) {
+            Io.printOutput(e.getMessage());
+        }
+    }
+
+    /**
+     * Parses the input and creates a new task based on the type of task
+     * 
+     * @param input
+     *              the input from the user
+     * @param type
+     *              the type of task
```

Comment:
The recommended style for comments is as follows:
@param x X coordinate of position.
 "@param input" and "the input from the user" should be in the same line.